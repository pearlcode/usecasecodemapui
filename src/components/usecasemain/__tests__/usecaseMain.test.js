/* eslint-disable react/jsx-filename-extension */
import { MemoryRouter as MRouter, Route } from 'react-router-dom';
import { React, render, mount } from '../../../__tests__/common/base';
import correctData from '../../../__mockdata__/correctUsecases';
import UsecaseMain from '../usecaseMain';

const usecaseListPath = '/test/usecases';
describe('usecaseMain', () => {
    it('should render the UsecaseMain', () => {
        const wrapper = render(
            <MRouter {...{ initialEntries: [usecaseListPath] }}>
                <Route
                    path="/usecases"
                    render={({ match }) => (
                        <UsecaseMain {...{ match }} usecases={correctData} />
                    )}
                />
            </MRouter>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should show form when click new UsecaseMain', () => {
        const wrapper = mount(
            <MRouter initialEntries={[usecaseListPath]}>
                <Route
                    path={`${usecaseListPath}`}
                    render={({ match }) => (
                        <UsecaseMain {...{ match }} usecases={correctData} />
                    )}
                />
            </MRouter>,
            {
                usecases: correctData,
            },
        );

        wrapper.find('button').simulate('click');
    });
});
