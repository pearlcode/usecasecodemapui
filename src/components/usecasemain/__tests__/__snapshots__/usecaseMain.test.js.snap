// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usecaseMain should render the UsecaseMain 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
